### PR TITLE
Update test acs-engine version

### PR DESCRIPTION
**Pull request by karataliu**

### CODE REVIEW

File: tests/k8s-azure/Dockerfile [hunk 1]
```diff
@@ -13,7 +13,7 @@ RUN git clone https://github.com/kubernetes/kubernetes . \
 RUN rm -rf _output/local/go
 
 FROM golang:1.8.3-stretch AS build_acs-engine
-ARG ACSENGINE_VERSION=v0.14.0
+ARG ACSENGINE_VERSION=v0.17.0
```
Comment (feiskyer) [inline tests/k8s-azure/Dockerfile]: we should always use latest release, the release tag could be get by command:

```
curl -L -s https://api.github.com/repos/Azure/acs-engine/releases/latest | jq -r .tag_name
```

Comment (karataliu) [reply tests/k8s-azure/Dockerfile]: Latest release sometimes have regression such as: https://github.com/Azure/acs-engine/issues/3022 ,which brings all NFS case to failure.

Would prefer manually updating the version, this is like what we do in using fixed os image version inside acs-engine.

Comment (feiskyer) [reply tests/k8s-azure/Dockerfile]: looks reasonable

**Pull request merged**